solutions/usaco-573.mdx

```diff
@@ -11,6 +11,115 @@ author: Qi Wang
 
 <LanguageSection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int number_of_wins(vector<int> &bessie, vector<int> &elsie, bool smaller_card_wins)
+{
+	/*
+	 * if person with larger value cards wins
+	 * Suppose person A and B use card with value 3 and 2 respectively
+	 * Person A wins as 3 > 2
+	 * But what if person with smaller value card wins (Second half of game)
+	 * We can say person B wins because (-2 > -3)
+	 * So we will just make all elements negative
+	 * instead of changing our way of comparing two numbers.
+	 */
+
+	int n = size(bessie);
+	assert(size(bessie) == size(elsie));
+
+	if (smaller_card_wins)
```

Comment:
how about instead of mulitiplying them by -1 you sort them differently instead?

Reply:
I actually thought of that but then user would have to know about functors, 
or you mean something like this `sort(a.begin(), a.end(), (condition ? greater<int>() : less<int>())`.

To avoid all that I used this simple trick
`2 > 3 => -2 < -3` so i can change the wya of comparing from greater to less easily

Reply:
a lot of c++ code in the guide uses custom stuff, don’t worry

Reply:
I added custom comparator instead of that trick.
It actually made the code shorter .. Thanks :)

Reply:
<img width="960" alt="image" src="https://user-images.githubusercontent.com/75680424/180299004-7c8e308a-b28c-4789-962b-0e71f1e5adf4.png">

Here is the AC screen for new code.